public/js/components/AtomEdit/CustomEditors/ChartEditor.js

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import Modal from '../../Utilities/Modal';
+import {atomPropType} from '../../../constants/atomPropType';
+
+export class ChartEditor extends React.Component {
+
+  static propTypes = {
+    atom: atomPropType.isRequired,
+    config: PropTypes.shape({
+      visualsUrl: PropTypes.string.isRequired,
+      stage: PropTypes.string.isRequired
+    }).isRequired
+  }
+
+  state = {
+    modalOpen: false
+  };
+
+  toggleModal = (e) => {
+    e.preventDefault();
+    if (this.state.modalOpen) {
+      this.closeModal();
+    } else {
+      this.openModal();
+    }
+  };
+
+  closeModal = () => {
+    this.setState({ modalOpen: false });
+    window.removeEventListener('message', this.onMessage, false);
+  };
+
+  openModal = () => {
+    this.setState({ modalOpen: true });
+    window.addEventListener('message', this.onMessage, false);
```

Comment:
`this.onMessage` doesn't exist?

Reply:
Whoops, copy and paste fail. It is now there